lib/src/widgets/yaru_window_control.dart

```diff
@@ -244,15 +244,24 @@ class _YaruWindowControlState extends State<YaruWindowControl>
   }
 
   Color _getIconColor(ColorScheme colorScheme) {
-    final iconColor = widget.iconColor?.resolve(_states);
-    if (iconColor != null) return iconColor;
+    final color = switch (style) {
+      YaruWindowControlPlatform.yaru => _getYaruIconColor(colorScheme),
+      YaruWindowControlPlatform.windows => _getWindowsIconColor(colorScheme)
+    };
 
-    switch (style) {
-      case YaruWindowControlPlatform.yaru:
-        return _getYaruIconColor(colorScheme);
-      case YaruWindowControlPlatform.windows:
-        return _getWindowsIconColor(colorScheme);
+    return color.withOpacity(interactive ? 1.0 : 0.5);
+  }
+
+  Color _getYaruIconColor(ColorScheme colorScheme) {
+    return widget.iconColor?.resolve(_states) ?? colorScheme.onSurface;
+  }
+
+  Color _getWindowsIconColor(ColorScheme colorScheme) {
+    var color = widget.iconColor?.resolve(_states) ?? colorScheme.onSurface;
+    if (interactive && _hovered && widget.type == YaruWindowControlType.close) {
+      color = Colors.white;
```

Comment:
```suggestion
    final color = widget.iconColor?.resolve(_states) ?? colorScheme.onSurface;
    if (interactive && _hovered && widget.type == YaruWindowControlType.close) {
      return Colors.white;
```

Reply:
Thanks!
Was moving the code back and forth until I found this brain fart (which I did some months ago):
https://github.com/ubuntu/yaru.dart/pull/889/files#diff-9a4c14a2bcd70496106f05b49dd6d3a4c2ec1e3f79055a51ee604f8220ea2181L231

🤦